scaladoc/src/dotty/tools/scaladoc/site/helpers/ConfigLoader.scala

```diff
@@ -0,0 +1,108 @@
+package dotty.tools.scaladoc.site.helpers
+
+import java.io.{File, InputStream}
+import java.nio.file.{Files, Paths, Path}
+import scala.jdk.CollectionConverters._
+import scala.collection.mutable
+import scala.language.dynamics
+import scala.language.dynamics
+import scala.language.dynamics
+import liqp.TemplateContext
+import liqp.nodes.LNode
+import liqp.tags.Tag
+import org.yaml.snakeyaml.Yaml
+
+class Config(private val data: mutable.LinkedHashMap[String, Any])
+    extends Dynamic {
+
+  def selectDynamic(field: String): Any = {
+    data.get(field) match {
+      case Some(value: mutable.LinkedHashMap[_, _]) =>
+        new Config(value.asInstanceOf[mutable.LinkedHashMap[String, Any]])
+      case Some(value: java.util.ArrayList[_]) =>
+        value.asScala.toSeq.map {
+          case map: java.util.Map[_, _] =>
+            new Config(
+              map
+                .asInstanceOf[java.util.Map[String, Any]]
+                .asScala
+                .to(mutable.LinkedHashMap)
+            )
+          case other => other
+        }
+      case Some(value) => value
+      case None => throw new NoSuchElementException(s"No such element: $field")
+    }
+  }
+
+  def get[T](key: String): Option[T] = {
+    data.get(key).map(_.asInstanceOf[T])
+  }
```

Comment:
an improvement here would be to use a scala.reflect.TypeTest implicit evidence: https://www.scala-lang.org/api/3.5.0/scala/reflect.html